frontend/src/Interfaces/APIResponses.tsx

```diff
@@ -0,0 +1,27 @@
+import Product from "./Product";
+
+export interface LoginResult {
+  userId: number | undefined;
+  accessToken: string | undefined;
+  error: string | undefined;
+}
```

Comment:
split into two responses instead

one for success - `LoginResponse`
one for error - `APIError`, which should be generic and applies to all APIs upon request failure
```js
export interface APIError {
  error: string
  details: string
}
```

Reply:
apply such pattern to other responses as well